riptide-capture/src/main/java/org/zalando/riptide/capture/DefaultCapture.java

```diff
@@ -19,8 +19,12 @@
 
     @Override
     public void tryAccept(@Nullable final T input) {
-        // TODO fail on second capture?
-        reference.compareAndSet(null, Optional.ofNullable(input));
+        final boolean captured = reference.compareAndSet(null, Optional.ofNullable(input));
+        final boolean alreadyCaptured = !captured;
```

Comment:
Not sure if this one improves readability, I would put negation into if condition

Reply:
Done